[NavigationBar] Vertically center navigationBar title within its default height for UIControlContentVerticalAlignmentTop

thkeen opened the pull request:
Since MDCNavigationBar is now using the iOS back button style, it doesn't look right to top align the title label. Update this -layoutSubviews implementation to use UIControlContentVerticalAlignmentCenter instead.

Doing an inspection shows that the title label is not currently center aligned with the back button.

### Thanks for starting a pull request on Material Components!

#### Don't forget:
- [x] Identify the component the PR relates to in brackets in the title. ```[Buttons] Updated documentation```
- [x] Link to GitHub issues it solves. ```closes #1596```
- [ ] Sign the CLA bot. You can do this once the pull request is submitted.

[Contributing](./contributing/README.md#pull-requests) has more information and tips for a great
pull request.

thkeen committed:
Update MDCNavigationBar titleVerticalAlignment to Center

Since MDCNavigationBar is now using the iOS back button style, it doesn't look right to top alignment the title label. Update this -layoutSubviews implementation to use UIControlContentVerticalAlignmentCenter instead.

ianegordon commented:
Need to verify with design.
Need to verify behavior with flexible header views.

ajsecord commented:
Thanks for the PR! What happens when you have a flexible header that is (much) taller than a single line of text? I think that's one of the issues.

thkeen commented:
Got it. I'll update the PR so it aligns vertically centered with the back button and still sticks to the top if the the flexible header is larger than 56.

thkeen committed:
Vertically center navigationBar title within default height

When UIControlContentVerticalAlignment is Top, still center it vertically within the default navigationBar height

thkeen committed:
Set originY to 0 if title height is larger than navigation bar

thkeen commented:
I've updated the changes accordingly.

thkeen commented:
CI complains "Unable to compare commits because the base of the compare did not upload a coverage report."

Can anyone guide how to generate a coverage report? Thanks.

thkeen commented:
Ping?

ianegordon commented:
Hi @thkeen.  I'm looking at this now.  I've kicked off a new Travis Build which will hopefully get us a coverage report.

ianegordon commented on components/NavigationBar/src/MDCNavigationBar.m:
```diff
@@ -448,7 +448,11 @@ - (CGRect)mdc_frameAlignedVertically:(CGRect)frame
-      return CGRectMake(frame.origin.x, bounds.origin.y, frame.size.width, frame.size.height);
+      CGFloat navigationBarCenteredY =
+          MDCFloor(([self intrinsicContentSize].height - frame.size.height) / 2);
+      navigationBarCenteredY = MAX(0, navigationBarCenteredY);
```
This seems surprising.  Why am I using centerY is I'm AlignmentTop.  Can you add a comment to clarify?  Something along the lines:
The title frame is vertically centered with the back button but will stick to the top of the header regardless of the header's height.

thkeen replied:
Done.

ianegordon reviewed (changes requested):
LGTM, just please add a comment clarifying what we are doing.

thkeen committed:
Add comment for back button vertically centered

When using UIControlContentVerticalAlignmentTop, vertically center to back button

thkeen reviewed (commented):
I've added the comments.

ianegordon commented:
The code coverage issue is unrelated to the PR.
Thank you @thkeen !
